lib/sles4sap/sap_deployment_automation_framework/deployment.pm

```diff
@@ -825,6 +826,30 @@ sub sdaf_ansible_verbosity_level {
     return '-vvvv';    # Default set to "-vvvv"
 }
 
+=head2 get_sdaf_instance_id
+
+    get_sdaf_instance_id(pattern=>['SCS', 'ERS', 'PAS']);
+
+Get instance id number from SAP_SYSTEM.tfvar.
+
+=over
+
+=item * B<pattern>: SDAF SAP Central Services pattern
+
+=back
+=cut
+
+sub get_sdaf_instance_id {
+    my (%args) = @_;
+    my $pattern = lc($args{pattern});
+    my $instance_id = '00';
+
+    my $tfvar_file = get_os_variable('sap_system_parameter_file');
+    $instance_id = script_output("grep ^${pattern}_instance_number $tfvar_file | cut -d '=' -f2 | grep -o '[0-9]\\+'");
```

Comment:
Is it possible to implement it more in Perl?

Pseudocode
```
$tfvar_content = script_output("cat $tfvar_file");
...
while (my $line = $tfvar_content) {
        if ($line =~ /^${pattern}_instance_number\s*=\s*([0-9]+)/) {
            ...
            split
            ...
            grep number
        }
    }
```

Reply:
Yes, I had thought about this way as well but in the end I quit as it needs more lines.